File: kinotic-js/workspace/packages/management-api/src/api/model/ProjectArtifacts.ts
import type { MicroserviceArtifact } from '@/api/model/MicroserviceArtifact'
import type { UiArtifact } from '@/api/model/UiArtifact'

/**
 * The artifacts one commit of a project contains, as the sync workload found them in the
 * checkout: what a deployment of that commit runs and publishes. Both lists are ordered by
 * name.
 */
export interface ProjectArtifacts {
    /**
     * The microservice artifacts, empty when the commit has none.
     */
    microservices: MicroserviceArtifact[]
    /**
     * The UI artifacts, empty when the commit has none.
     */
    uis: UiArtifact[]
}

File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/ProjectArtifacts.java
package org.kinotic.management.api.model;

import java.util.List;

/**
 * The artifacts one commit of a project contains, as the sync workload found them in the
 * checkout: what a deployment of that commit runs and publishes. Both lists are ordered by
 * name.
 *
 * @param microservices the microservice artifacts, empty when the commit has none
 * @param uis           the UI artifacts, empty when the commit has none
 */
public record ProjectArtifacts(List<MicroserviceArtifact> microservices, List<UiArtifact> uis) {}
